### PR TITLE
Give testset a name

### PR DESCRIPTION
**Pull request by jeroensmit**

### CODE REVIEW

**Comment by TomasVotruba** [comment]
What's the reason behind this?

**Comment by jeroensmit** [comment]
Is easier to spot which example file is causing exceptions

PHPunit output:

before change:
Rector\DeadCode\Tests\Rector\Class_\RemoveSetterOnlyPropertyAndMethodCallRector\RemoveSetterOnlyPropertyAndMethodCallRectorTest::test with data set #0 

after change:
Rector\DeadCode\Tests\Rector\Class_\RemoveSetterOnlyPropertyAndMethodCallRector\RemoveSetterOnlyPropertyAndMethodCallRectorTest::test with data set "deal_with_property_fetches.php.inc"

**Comment by TomasVotruba** [comment]
How would you re-run only the file?

**Comment by jeroensmit** [comment]
the complete name filter filters on looks something like:
Rector\Architecture\Tests\Rector\Class_\ConstructorInjectionToActionInjectionRector\ConstructorInjectionToActionInjectionRectorTest::test with data set "duplicate.php.inc"

The filter option works as an regex so you could do:

```bash
vendor/bin/phpunit --filter ConstructorInjectionToAction.*duplicate
```
or

```bash
vendor/bin/phpunit --filter ConstructorInjectionToAction --filter duplicate
```

a part of the file names suffices (doesn't need to be from te start)

**Comment by TomasVotruba** [comment]
How would you compare it to `--filter test#1`

**Comment by TomasVotruba** [comment]
Having both options would be something I'd merge:

```bash
Rector\DeadCode\Tests\Rector\Class_\RemoveSetterOnlyPropertyAndMethodCallRector\RemoveSetterOnlyPropertyAndMethodCallRectorTest::test with data set "deal_with_property_fetches.php.inc", #0   
```

**Comment by jeroensmit** [comment]
Just what I was thinking.

Sadly the best thing I can reach is this:

`vendor/bin/phpunit --filter RemoveSetter@1.*`

If you find that sufficient I'll update the pullrequest. 
Otherwise decline. I can live with that ;-)

**Comment by TomasVotruba** [comment]
Yes, that's what I meant :) go for it

File: src/Testing/StaticFixtureProvider.php [hunk 1]
```diff
@@ -14,8 +14,10 @@ public static function yieldFilesFromDirectory(string $directory, string $suffix
     {
         $fileInfos = self::findFilesInDirectory($directory, $suffix);
 
+        $key = 0;
         foreach ($fileInfos as $fileInfo) {
-            yield $fileInfo->getFilename() => [$fileInfo->getPathName()];
+            yield "{$key} " . $fileInfo->getFilename() => [$fileInfo->getPathName()];
+            $key++;
```
Comment (TomasVotruba) [inline src/Testing/StaticFixtureProvider.php]: Why not use key of this foreach?

Comment (jeroensmit) [reply src/Testing/StaticFixtureProvider.php]: key contains the filepath

Comment (TomasVotruba) [reply src/Testing/StaticFixtureProvider.php]: Ah, I used `FinderSanitizer` to remove this WTF

Ok

**Comment by TomasVotruba** [comment]
Thank you

**Comment by TomasVotruba** [comment]
I'm finally testing it an it breaks filtering by provider name `test#1`

```bash
vendor/bin/phpunit packages/PHPUnit/tests/Rector/MethodCall/ReplaceAssertArraySubsetRector/ReplaceAssertArraySubsetRectorTest.php --filter test#1 
```

**Comment by jeroensmit** [comment]
Sorry for taking a while.

Hopefully you managed by now, if not the solution was using @ instead of #
test@1.*

**Pull request merged**